firestore/src/common/settings.cc

```diff
@@ -70,6 +121,40 @@ std::ostream& operator<<(std::ostream& out, const Settings& settings) {
   return out << settings.ToString();
 }
 
+bool operator==(const Settings& lhs, const Settings& rhs) {
+  if (lhs.host() != rhs.host()) {
+    return false;
+  }
+
+  if (lhs.is_ssl_enabled() != rhs.is_ssl_enabled()) {
+    return false;
+  }
+
+  if (lhs.cache_settings_source_ != rhs.cache_settings_source_) {
```

Comment:
What if the settings sources are different but represent the same cache configuration? Do the other SDKs consider this "equal" or "unequal"? We should be consistent with the other SDKs here.